src/test/java/core/basesyntax/service/ParserTest.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.dto.Transaction;
+import core.basesyntax.model.Fruit;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ParserTest {
+    private static final String PURCHASE = "p";
+    private static final Fruit BANANA = new Fruit("banana");
+    private static Parser parser;
+
+    @BeforeClass
+    public static void before() {
+        parser = new ParserImpl();
+    }
+
+    @Test
+    public void parseLine_ValidLine_Ok() {
+        String lineToParse = "p,banana,2019";
+        Transaction expected = new Transaction(PURCHASE, BANANA, 2019);
+        Transaction actual = parser.parseLine(lineToParse);
+        assertEquals(expected.getFruit(), actual.getFruit());
+        assertEquals(expected.getOperation(), actual.getOperation());
+        assertEquals(expected.getQuantity(), actual.getQuantity());
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void parseLine_InvalidLine_NotEnoughElements_Not_Ok() {
+        String lineToParse = "p,fd,";
+        parser.parseLine(lineToParse);
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void parseLine_InvalidLine_NotNumber_Not_Ok() {
+        String lineToParse = "p,fd,number";
+        parser.parseLine(lineToParse);
+    }
+
+    @Test
+    public void parseLine_InvalidLine_Not_Ok() {
+        String lineToParse = "psdf,fdsdfadfgdsa,1234";
```

Comment:
this method name is a bit confusing) if we don't get exception, but our program allowes to pass such a string - let's name test `Ok`